pkg/view/component/table.go

```diff
@@ -287,7 +287,7 @@ func (t *Table) AddButton(name string, payload action.Payload, buttonOptions ...
 		t.Config.ButtonGroup = NewButtonGroup()
 	}
 	button := NewButton(name, payload, buttonOptions...)
-	t.Config.ButtonGroup.AddButton(button)
+	t.Config.ButtonGroup.AddButton(*button)
```

Comment:
This is an inconsistent API experience. The caller should not have to have knowledge of how to transform the button to pass it in to `AddButton` .. `NewButton` should return the type that `AddButton` expects and then we can remove the `*`

Reply:
You are absolutely right, I just made the update for the `AddButton` method